### PR TITLE
Relate TUF design to Notary v2 goals

### PR DESCRIPTION
**Pull request by mnm678**
Based on discussion in the last meeting, I added a section to clarify why the TUF design is being proposed that puts on paper some of what has previously been discussed in meetings about the use of TUF in Notary v2.

### CODE REVIEW

**Review by SteveLasker — COMMENTED** [review]
Thanks @mnm678 for the updates. I've added comments that reflect the original notes, as the addition of the requirements & goals are helpful.

File: tuf-design.md [hunk 1]
```diff
@@ -4,6 +4,36 @@ Based on discussions with the Notary v2 community over the past few months, we d
 
 This design will be used in conjunction with the OCI artifact work done by others in the Notary v2 community to ensure compliance with the OCI specification and integration with existing registries.
 
+## Alignment with Notary v2 project goals
+
+The use of TUF in Notary v2 is part of an effort to address the [goals](https://github.com/notaryproject/requirements#goals) of the Notary v2 project.
```
Comment (NiazFK) [inline tuf-design.md:9]: Could we also address the requirements that have been agreed upon in key management?

**Comment by mnm678** [comment]
@SteveLasker Thanks for the review! I added some clarifications inline to address many of your comments.

> > Updating fully qualified references to artifacts
> 
> The initial signature should be fully intact. Including what registry it originated from. No change to the content, or the wraper. This is the integrity check.

In this case, what would a registry do if they want to maintain this original signature as well as re-sign it? Should they re-sign using the original fully qualified reference? I can see two possible options here, in one the signature is copied and the original signer is the primary party trusted to sign the image. In this case any additional signatures would have to match the original name so they could have an exact match. In the other, the artifact is being copied into a private registry and the copier is the party trusted to sign it. In this case, it may be desirable to sign the image using the fully qualified reference in the new registry.

There may also be an in-between option that allows only the last part of the name to match, but this raises questions about the scope of namespaces.

> 
> > Hiding images from other users
> 
> This section implies Coke & Pepsi content can be aggregated in some way, even at some anonymized leve is not permitted.
> 
> It's great to see the requirements get added and covered. The concerns over immutability, and how much updating of content must be maintained is of great concern.
> 
> Consider we might [etched all the immutable content pushed to a registry in glass (Project Silica)](https://news.microsoft.com/innovation-stories/ignite-project-silica-superman/). The basic premise of TUF suggests a continual service runs to continually update snapshot and timestamps of all data. The vast majority of registry content is immutable. Following [best tagging practices](https://stevelasker.blog/2018/03/01/docker-tagging-best-practices-for-tagging-and-versioning-docker-images/), even the tags should be unique, with the important exception of stable tags. And, we do need a solution for tag signing/locking. However, I'd suggest we should be focusing on a solution that doesn't require all data must be continually updated to assure it's still valid. It's a non-scalable solution. If the computers stop processing the data, because the data was etched in glass, put in a mountain, revived years later, is that data considered invalid?

Users who need to download an image today have different security requirements than users who want to download an image from a mountain years later. Users today need a way to ensure that the signature they are verifying was signed by a key that was valid today, and not compromised or expired. This is vital to any mechanism that prioritizes key revocation. Users years from now have the benefit of hindsight to avoid compromised packages, and so can ignore timestamps when verifying metadata. I would argue that it is more important to ensure that users today are downloading secure software than it is to ensure that future users don’t have to override a timestamp check when using legacy software.

File: tuf-design.md (hunk 1 above)
Comment (mnm678) [reply tuf-design.md:9]: Added!

**Comment by mnm678** [comment]
I'm going to close this in favor of a discussion of goals on the [tuf-notary](https://github.com/notaryproject/tuf) repository.

**Pull request closed without merging**